srcStatic/FileSystemHelper.cpp

```diff
@@ -42,14 +42,12 @@ bool Exists(const std::string& path)
 	return fs::exists(modifiedPath);
 }
 
-std::vector<std::string> LocateFilesWithExtension(const std::string& directory, std::string extension)
+std::vector<std::string> FindFilesWithExtension(const std::string& basePath, const std::string& relativeSearchPath, std::string extension)
 {
-	if (!IsDirectory(directory)) {
-		return {};
-	}
+	auto directory = fs::path(basePath) / relativeSearchPath;
 
-	if (extension.empty()) {
-		throw std::runtime_error("provided extension cannot be empty");
+	if (!IsDirectory(directory.string())) {
+		return {};
```

Comment:
Would it be better to throw an exception here to indicate that improper data was provided?

Reply:
Technically that aspect isn't really addressed in this PR. It just shows up as a highlighted change due to changes to surrounding lines. The diff is a bit confused here.

I'm fine with either way. I think I have a slight preference for returning an empty list (as the code previously did) rather than throwing an exception. If you try to find files in a non-existent directory, there are of course no such files to be found.

Reply:
Ok, let's just return empty for non-existent directory.